lib/foreman/renderer/base_template_scope/base_variables.rb

```
@@ -0,0 +1,125 @@
+module Foreman
+  module Renderer
+    class BaseTemplateScope
+      class BaseVariables
+        @@loaders = %w(load_template_vars)
+
+        def initialize(host, params = {})
+          @host = host
+          @params = params
+          @loaded = []
+        end
+
+        def load_variables
+          loaders.each do |loader|
+            if respond_to?(loader, true) && !loaded.include?(loader)
+              send(loader)
+              loaded << loader
+            end
+          end
+        end
+
+        def to_hash
+          instance_variables.map do |key|
+            [key.to_s[1..-1].to_sym, instance_variable_get(key)]
+          end.to_h
+        end
+
+        def self.register_loader(loader_name)
+          @@loaders.push(loader_name) unless @@loaders.include?(loader_name)
+        end
+
+        private
+
+        attr_reader :host, :params
+        attr_accessor :loaded
+
+        delegate :diskLayout, :medium, :architecture, :ptable, :use_image, :arch, :image_file,
+                 :default_image_file, to: :host, allow_nil: true
```

is `:allow_nil` a good idea? I mean this will fail silently if rendered for nil host, or more probable scenario, host's OS is nil (line below), would it be better to have these defined explicitly and use `check_host` so that it raises proper renderer exception in case host is not available?

I see you're defininig operatingsystem below, so perhaps defining `host` method the same way and adding `check_host` there would be best.